### PR TITLE
Adding LaTeX with MathJaX

### LINKED ISSUE #253: No tooltip on DeleteButton
Right now, there is no tool tip on the DeleteButton for pages (but for new page button)

### PR DESCRIPTION
**Pull request by acolombier**
Fixes #253: No tooltip on DeleteButton

Changes made in this pull request: 
- LaTeX support with MathJaX (`$$<equation>$$` for block, `\(<equation>\)` for inline)
- LaTeX coloration

### CODE REVIEW

File: src/Widgets/Viewer.vala [hunk 1]
```diff
@@ -200,6 +200,7 @@ public class ENotes.Viewer : WebKit.WebView {
         mkd.document (out result);
 
         string html = "<!doctype html><meta charset=utf-8><head>";
+        html += "<script type=\"text/javascript\" async src=\"/usr/share/notes-up/mathjax/MathJax.js?config=TeX-MML-AM_CHTML\"></script>";
```
Comment (Philip-Scott) [inline src/Widgets/Viewer.vala]: You should not be adding this script on every page. only when it's being used in the plugin

File: src/Widgets/PagesList.vala [hunk 2]
```diff
@@ -118,6 +118,7 @@ public class ENotes.PagesList : Gtk.Box {
         plus_button.can_focus = false;
         minus_button.no_show_all = true;
         separator.no_show_all = true;
+        minus_button.set_tooltip_text (_("Delete Page"));
```
Comment (Philip-Scott) [inline src/Widgets/PagesList.vala]: This should be moved to it's own PR.

A single PR should only change/add one feature at a time :)

File: src/Widgets/Viewer.vala (hunk 1 above)
Comment (acolombier) [reply src/Widgets/Viewer.vala]: Okay, but unfortunately, MathJaX seems not to be really happy when you load it in the body, and produce some weird behavior sometime especially for inline notations

File: src/Widgets/PagesList.vala (hunk 2 above)
Comment (acolombier) [reply src/Widgets/PagesList.vala]: Alright, I didn't know about that :/

**Comment by acolombier** [comment]
Although I completely understand the point, I am affraid you will not have any other way if you want to add LaTeX support. Furthermore, if you diff the code, you will see that it is exactly the latest mathjax code coming from its official repo, minus fews dozen of MB (mostly images) provided for retarded browser, completely not required in the current webview.

File: src/Widgets/PagesList.vala (hunk 2 above)
Comment (Philip-Scott) [reply src/Widgets/PagesList.vala]: No problem :) Just helping you learn the best practices for when you contribute in another code

**Comment by kostrykin** [comment]
Maybe you @acolombier should consider creating a fork if this project with the LaTeX support. Personally, I would appreciate this a lot.

**Comment by acolombier** [comment]
@kostrykin, I already forked the project for the implementation! It would actually be amazing if you could give me a feedback on the current implementation!

**Comment by Philip-Scott** [comment]
I mean, I someone were to find a way to implement this without adding 30,000 lines on unknown JS code, I wouldn't be against adding support for the feature. For now though, I'll be closing this PR :) 

I hope you understand why doing it this way is not a good practice

**Comment by Philip-Scott** [comment]
@acolombier @kostrykin, here's a very good read of why adding unknown JS (or really, any code) is a bad idea: 

https://hackernoon.com/im-harvesting-credit-card-numbers-and-passwords-from-your-site-here-s-how-9a8cb347c5b5

**Comment by acolombier** [comment]
@Philip-Scott thank you for this interesting lecture, even thought what it relates about is something that I know well. However, in the world of the Open Source, it is required sometimes to trust the work of the other if you want to be able to build something interesting. For instance, how many websites use jQuery ? And on these websites, how many developers have lost 3 weeks reading the code before putting it on there websites ? 

Furthermore here, we are not talking about the work of few developer from the shadow, but a big project well spread out and supported. Finally, even if I understand and accept your decision, I think it is important to scale the risk that this blind integration could bring : I don't think any of the users is using Notes Up to store credentials information, and thus, the rob that a rogue JS could occur would just be some uni lecture notes in my case.
